### DISCUSSION TITLE
NGINX | File not found.

### DISCUSSION THREAD

**Discussion by ghost** [osm-search/Nominatim#2861] [General]
Hello, I just finished full planet import and I am now faced with deploying my webserver on Nginx, for that I followed the tutorial on Ubuntu 22 : https://nominatim.org/release-docs/latest/appendix/Install-on-Ubuntu-22/

Issue is when I go to the external IP of my machine, I receive a "File not found." on the page.

I have read it might have to do something with `fastcgi_param` but I am not sure.

My webserver files are located in `/srv/nominatim/website`

my conf files are:

```
[www]
; Replace the tcp listener and add the unix socket
listen = /var/run/php8.1-fpm.sock

; Ensure that the daemon runs as the correct user
listen.owner = www-data
listen.group = www-data
listen.mode = 0666

; Unix user of FPM processes
user = www-data
group = www-data

; Choose process manager type (static, dynamic, ondemand)
pm = ondemand
pm.max_children = 5

```

```
server {
    listen 80 default_server;
    listen [::]:80 default_server;

    root /srv/nominatim/website;
    index search.php index.html;
    location / {
        try_files $uri $uri/ @php;
    }

    location @php {
        fastcgi_param SCRIPT_FILENAME "$document_root$uri.php";
        fastcgi_param PATH_TRANSLATED "$document_root$uri.php";
        fastcgi_param QUERY_STRING    $args;
        fastcgi_pass unix:/var/run/php8.1-fpm.sock;
        fastcgi_index index.php;
        include fastcgi_params;
    }

    location ~ [^/]\.php(/|$) {
        fastcgi_split_path_info ^(.+?\.php)(/.*)$;
        if (!-f $document_root$fastcgi_script_name) {
            return 404;
        }
        fastcgi_pass unix:/var/run/php7.4-fpm.sock;
        fastcgi_index search.php;
        include fastcgi.conf;
    }
}

```

**Comment by mtmail** [comment]
I see `php7.4-fpm.sock` mentioned. I think that should be 8.1. Might be a mistake in the documentation.

>     root /nominatim-project/website;

I think that should be `root /srv/nominatim/nominatim-project/website`

Check the logfiles of `/var/log/nginx/access.log`, `/var/log/nginx/error.log` and `/var/log/php8.1-fpm.log`

Check that e.g. the `status.php` has a line `@define('CONST_LibDir', ...` near the top. If that's missing then you're looking at the source file. `nominatim refresh --website` converts the source files into usable PHP files by adding the `@define` lines.

**Reply by ghost** [reply]
> > ```
> > root /nominatim-project/website;
> > ```

I changed the root to the correct folder, for me it is  `/srv/nominatim/website` even though I have seen the documentation a discrepancy with  root `/nominatim-project/website;`

https://nominatim.org/release-docs/latest/appendix/Install-on-Ubuntu-22/#option-2-using-nginx : Calls for `nominatim` as USERHOME

https://nominatim.org/release-docs/latest/admin/Deployment/: Calls for `nominatim-project` as USERHOME

I updated the following: 

- my php8.1-fpm was indeed inside /var/run/php so I had to modify the path.
- changed `php7.4-fpm.sock` to `php8.1-fpm.sock`

Now I get a 502 Bad Gateway (I guess its progress)
```
server {
    listen 80 default_server;
    listen [::]:80 default_server;

    root /srv/nominatim/website;
    index search.php index.html;
    location / {
        try_files $uri $uri/ @php;
    }

    location @php {
        fastcgi_param SCRIPT_FILENAME "$document_root$uri.php";
        fastcgi_param PATH_TRANSLATED "$document_root$uri.php";
        fastcgi_param QUERY_STRING    $args;
        fastcgi_pass unix:/var/run/php/php8.1-fpm.sock;
        fastcgi_index index.php;
        include fastcgi_params;
    }

    location ~ [^/]\.php(/|$) {
        fastcgi_split_path_info ^(.+?\.php)(/.*)$;
        if (!-f $document_root$fastcgi_script_name) {
            return 404;
        }
        fastcgi_pass unix:/var/run/php/php8.1-fpm.sock;
        fastcgi_index search.php;
        include fastcgi.conf;
    }
}
```

**Reply by ghost** [reply]
/var/log/nginx/access.log :

```
"GET /search.php?q=Berlin HTTP/1.1" 404 197 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/106.0.0.0 Safari/537.36"
```


/var/log/nginx/error.log:
```
[crit] 140116#140116: *1 connect() to unix:/var/run/php/php8.1-fpm.sock failed (2: No such file or directory) while connecting to upstream, client: 78.198.152.149, server: , request: "GET / HTTP/1.1", upstream: "fastcgi://unix:/var/run/php/php8.1-fpm.sock:", host: "34.148.200.182"
```


/var/log/php8.1-fpm.log:
```
[03-Nov-2022 21:29:35] NOTICE: fpm is running, pid 139612
[03-Nov-2022 21:29:35] NOTICE: ready to handle connections
[03-Nov-2022 21:29:35] NOTICE: systemd monitor interval set to 10000ms
[03-Nov-2022 21:40:46] NOTICE: Terminating ...
[03-Nov-2022 21:40:46] NOTICE: exiting, bye-bye!
[03-Nov-2022 21:40:46] NOTICE: fpm is running, pid 139832
[03-Nov-2022 21:40:46] NOTICE: ready to handle connections
[03-Nov-2022 21:40:46] NOTICE: systemd monitor interval set to 10000ms
[03-Nov-2022 21:42:12] NOTICE: Terminating ...
[03-Nov-2022 21:42:12] NOTICE: exiting, bye-bye!
[03-Nov-2022 21:42:12] NOTICE: fpm is running, pid 139898
[03-Nov-2022 21:42:12] NOTICE: ready to handle connections
[03-Nov-2022 21:42:12] NOTICE: systemd monitor interval set to 10000ms
[03-Nov-2022 21:45:01] NOTICE: Terminating ...
[03-Nov-2022 21:45:01] NOTICE: exiting, bye-bye!
[03-Nov-2022 21:45:01] NOTICE: fpm is running, pid 139992
[03-Nov-2022 21:45:01] NOTICE: ready to handle connections
[03-Nov-2022 21:45:01] NOTICE: systemd monitor interval set to 10000ms
[03-Nov-2022 21:45:09] NOTICE: Terminating ...
[03-Nov-2022 21:45:09] NOTICE: exiting, bye-bye!
[03-Nov-2022 21:45:09] NOTICE: fpm is running, pid 140048
[03-Nov-2022 21:45:09] NOTICE: ready to handle connections
[03-Nov-2022 21:45:09] NOTICE: systemd monitor interval set to 10000ms
[03-Nov-2022 21:46:36] NOTICE: Terminating ...
[03-Nov-2022 21:46:36] NOTICE: exiting, bye-bye!
[03-Nov-2022 21:46:36] NOTICE: fpm is running, pid 140109
[03-Nov-2022 21:46:36] NOTICE: ready to handle connections
[03-Nov-2022 21:46:36] NOTICE: systemd monitor interval set to 10000ms
```

**Reply by ghost** [reply]
Okay seems fixed, the php-fm file didn't have the version attached to it, is that normal you think?
<img width="343" alt="Capture d’écran 2022-11-03 à 22 57 00" src="https://user-images.githubusercontent.com/92714362/199842131-ce219ccb-d6cb-4db3-a45a-8e6562df6174.png">

**Reply by ghost** [reply]
Also I want to test out my server using the `geopy` module, do you know by any chance if I can make it use http (port 80) instead of https (443)?

`geopy.exc.GeocoderUnavailable: HTTPSConnectionPool(host='http', port=443): Max retries exceeded with url `

**Reply by mtmail** [reply]
I don't have the version attached on my files either. So at least one error in the documentation. I'll update that. We do run nginx as part of the test-suite, but only the older Ubuntu-18 one https://github.com/osm-search/Nominatim/blob/master/.github/workflows/ci-tests.yml#L177

**Reply by mtmail** [reply]
geopy: No idea

**Reply by ghost** [reply]
And I guess there is no tutorials in the Nominatim doc to setup SSL on Nginx?

**Reply by mtmail** [reply]
How to setup nginx with SSL is outside the scope of Nominatim support
